Flaky Test: TestChunksEnqueueFailed

[example failure](https://github.com/apache/pulsar-client-go/actions/runs/4349818338/jobs/7601251635)

<details>
<summary>Log</summary>
<pre><code>
=== RUN   TestChunksEnqueueFailed
time="2023-03-07T04:08:35Z" level=info msg="Connecting to broker" remote_addr="pulsar://localhost:6650"
time="2023-03-07T04:08:35Z" level=info msg="TCP connection established" local_addr="127.0.0.1:58584" remote_addr="pulsar://localhost:6650"
time="2023-03-07T04:08:35Z" level=info msg="Connection is ready" local_addr="127.0.0.1:58584" remote_addr="pulsar://localhost:6650"
time="2023-03-07T04:08:35Z" level=info msg="Connected producer" cnx="127.0.0.1:58584 -> 127.0.0.1:6650" epoch=0 topic="persistent://public/default/my-topic-655099802"
time="2023-03-07T04:08:35Z" level=info msg="Created producer" cnx="127.0.0.1:58584 -> 127.0.0.1:6650" producerID=1 producer_name=standalone-0-429 topic="persistent://public/default/my-topic-655099802"
    message_chunking_test.go:297: 
        	Error Trace:	/pulsar/pulsar-client-go/pulsar/message_chunking_test.go:297
        	Error:      	An error is expected but got nil.
        	Test:       	TestChunksEnqueueFailed
    message_chunking_test.go:298: 
        	Error Trace:	/pulsar/pulsar-client-go/pulsar/message_chunking_test.go:298
        	Error:      	Expected nil, but got: &pulsar.chunkMessageID{messageID:(*pulsar.messageID)(0xc000ea6760), firstChunkID:(*pulsar.messageID)(0xc000bd1e60), receivedTime:time.Time{wall:0x0, ext:0, loc:(*time.Location)(nil)}, consumer:pulsar.acker(nil)}
        	Test:       	TestChunksEnqueueFailed
time="2023-03-07T04:08:35Z" level=info msg="Closing producer" producerID=1 producer_name=standalone-0-429 topic="persistent://public/default/my-topic-655099802"
time="2023-03-07T04:08:35Z" level=info msg="Closed producer" producerID=1 producer_name=standalone-0-429 topic="persistent://public/default/my-topic-655099802"
--- FAIL: TestChunksEnqueueFailed (0.08s)
</code></pre>
</details>

--- a/pulsar/message_chunking_test.go
+++ b/pulsar/message_chunking_test.go
@@ -22,6 +22,7 @@ import (
 	"errors"
 	"fmt"
 	"math/rand"
+	"net/http"
 	"strings"
 	"sync"
 	"testing"
@@ -290,6 +291,13 @@ func TestChunksEnqueueFailed(t *testing.T) {
 	assert.NoError(t, err)
 	assert.NotNil(t, producer)
 	defer producer.Close()
+
+	// Reduce publish rate to prevent the producer sending messages too fast
+	url := adminURL + "/" + "admin/v2/persistent/public/default/" + topic + "/publishRate"
+	makeHTTPCall(t, http.MethodPost, url, "{\"publishThrottlingRateInMsg\": 1,\"publishThrottlingRateInByte\": 1000}")
+
+	// Need to wait some time to let the rate limiter take effect
+	time.Sleep(2 * time.Second)
 
 	ID, err := producer.Send(context.Background(), &ProducerMessage{
 		Payload: createTestMessagePayload(1000),

